x/yarpctest/request_unary.go

```diff
@@ -168,6 +173,13 @@ func GiveTimeout(duration time.Duration) api.RequestOption {
 	})
 }
 
+// UnaryOutboundMiddleware sets unary outbound middleware for a request.
+func UnaryOutboundMiddleware(out ...middleware.UnaryOutbound) api.RequestOption {
+	return api.RequestOptionFunc(func(opts *api.RequestOpts) {
+		opts.UnaryMiddleware = out
```

Comment:
consider appending here instead in case unary outbound middleware is specified multiple times.